crates/bevy_ecs/src/observer/system_param.rs

```diff
@@ -106,23 +106,19 @@ impl<'w, E, B: Bundle> On<'w, E, B> {
 }
 
 impl<'w, E: EntityEvent, B: Bundle> On<'w, E, B> {
-    /// Returns the [`Entity`] that was targeted by the `event` that triggered this observer.
+    /// Returns the [`Entity`] that was targeted by the [`EntityEvent`] that triggered this observer.
     ///
     /// Note that if event propagation is enabled, this may not be the same as the original target of the event,
     /// which can be accessed via [`On::original_target`].
-    ///
-    /// If the event was not targeted at a specific entity, this will return [`Entity::PLACEHOLDER`].
     pub fn target(&self) -> Entity {
-        self.trigger.current_target.unwrap_or(Entity::PLACEHOLDER)
+        self.trigger.current_target.unwrap()
```

Comment:
Follow-up: we can probably avoid storing an Option and then unwrapping here at all. I think that will require decoupling the internals for entity-events vs broadcast-events.

Reply:
Agreed, I chose to make the minimal changes for now and updated the docs